tests/src/test/scala/mainecoon/tests/autoFlatMapTests.scala

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2017 Kailuo Wang
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package mainecoon.tests
+
+import cats.FlatMap
+import cats.instances.function._
+import cats.kernel.Eq
+import cats.kernel.instances.string._
+import cats.kernel.instances.tuple._
+import cats.laws.discipline.eq._
+import cats.laws.discipline.{FlatMapTests, SerializableTests}
+import cats.syntax.arrow._
+import mainecoon.autoFlatMap
+import mainecoon.tests.autoFlatMapTests._
+import org.scalacheck.{Arbitrary, Cogen}
+
+class autoFlatMapTests extends MainecoonTestSuite {
+
+  checkAll("FlatMap[TestAlg]", FlatMapTests[TestAlgebra].flatMap[Float, String, Int])
+  checkAll("FlatMap[SimpleAlg]", SerializableTests.serializable(FlatMap[TestAlgebra]))
```

Comment:
since you replaced `SimpleAlg` with `TestAlgebra`, would you change the text here as well?